cmd/object-api-utils.go

```diff
@@ -163,6 +163,29 @@ func pathJoin(elem ...string) string {
 	return path.Join(elem...) + trailingSlash
 }
 
+// pathClean - is path.Clean() but retains trailing "/" and returns
+// empty string for empty strings not ".".
+func pathClean(p string) string {
+	dotdot := strings.HasPrefix(p, "..")
+	if dotdot {
+		p = slashSeparator + p
+	}
+	// Check path ends with separate, save it.
+	sp := hasSuffix(p, slashSeparator)
+	p = path.Clean(p)
+	if p == "." {
```

Comment:
We are cleaning object names with pathClean() function, if we remove '.', it means we are not able to upload an object with name '.' though this is already supported in S3.